core/src/main/java/com/sequenceiq/cloudbreak/service/stackpatch/UnboundRestartPatchService.java

```diff
@@ -75,15 +80,21 @@ public boolean isAffected(Stack stack) {
     @Override
     void doApply(Stack stack) throws ExistingStackPatchApplyException {
         if (isCmServerReachable(stack)) {
-            FlowIdentifier flowIdentifier = ThreadBasedUserCrnProvider.doAsInternalActor(() -> clusterOperationService.updateSalt(stack));
+            FlowIdentifier flowIdentifier = ThreadBasedUserCrnProvider.doAs(
+                    internalCrnModifier.getInternalCrnWithAccountId(Crn.fromString(stack.getResourceCrn()).getAccountId()),
+                    () -> clusterOperationService.updateSalt(stack));
+            LOGGER.debug("Starting update salt for stack {} with flow {}", stack.getResourceCrn(), flowIdentifier.getPollableId());
             Boolean success = Polling.waitPeriodly(1, TimeUnit.MINUTES)
                     .run(() -> pollFlowState(flowIdentifier));
             if (!success) {
-                throw new ExistingStackPatchApplyException("Failed to update salt for stack " + stack.getResourceCrn());
+                String message = String.format("Failed to update salt for stack %s with flow %s", stack.getResourceCrn(), flowIdentifier.getPollableId());
+                LOGGER.warn(message);
+                throw new ExistingStackPatchApplyException(message);
```

Comment:
just a minor thing: this logic appers twice in this method, maybe worth to extract it into a different method like "handleErrorDuringApply" or something like that